### DISCUSSION TITLE
Metamask wallet regression: eth_sendTransaction always ends with "Invalid Id"

### DISCUSSION THREAD

**Discussion by Sabriel1984** [WalletConnect/walletconnect-monorepo#5143] [Developer Support]
I have Android and iOS apps with web3modal library and send request with eth_sendTransaction just stopped working without any reason - I don't know if it's metamask or wallet connect problem.

I have latest versions of wallet-connect bom and metamask.

Request:
WalletManagerImpl sendTransactionRequest success WalletConnect(method=eth_sendTransaction, params=[{"from":"0xfba2964efc69565d58b3429254db3225cf701035","to":"0x4BDDf9D49F7bE9957f60B5ce17ee32cf3d4A2107","value":"0x0", "data":"0x095ea7b3000000000000000000000000150b1ea321a42a934cbc970a7cf015dbac34b752000000000000000000000000000000000000000000000008e962295c87f43d30"}], chainId=eip155:11155111, requestId=1719577549980500, sessionTopic=a15dc9097a891024c2a53da98ed4fcb8fcc1f88df784b12e3ce40bbdab4a837f)

so basically I use eth_sendTransaction with only from, to and data filled (value is always 0 and it's not required) - becuase we try to send our token (by dapp with wallet connect library) from client metamask wallet to our smart contract address - so I have smart contract abi encoded and filled with values in "data" field

And it used to works lets say often in previous wallet versions (so in most cases dialog appears - if not there was no error in logs), but on latest 7.24 version it completely stop working - and I always have this error in logs:

Response:
WalletManagerImpl onSessionRequestResponse SessionRequestResponse(topic=a15dc9097a891024c2a53da98ed4fcb8fcc1f88df784b12e3ce40bbdab4a837f, chainId=eip155:11155111, method=eth_sendTransaction, result=JsonRpcError(id=1719577549980500, code=1, message=Invalid Id))

I checked and for the same params by web browser - https://docs.metamask.io/wallet/reference/eth_sendtransaction - it seems to work perfectly - so it must by regression in mobile metamask app. I also use another wallet (DeFi crypto.com) and the same transaction works without any problems.

**Comment by glitch-txs** [comment]
This is a MetaMask issue, could you contact them instead? Thanks!
